go/libraries/doltcore/sqle/binlogreplication/binlog_type_serialization.go

```diff
@@ -291,10 +292,17 @@ func (d decimalSerializer) serialize(_ *sql.Context, typ sql.Type, descriptor va
 				decimalValue.Exponent(), decimalValue.String())
 		}
 
-		absStringVal := decimalValue.Abs().String()
-		firstFractionalDigitIdx := len(absStringVal) + int(decimalValue.Exponent())
-		stringIntegerVal := absStringVal[:firstFractionalDigitIdx-1]
-		stringFractionalVal := absStringVal[firstFractionalDigitIdx:]
+		absStringVal := decimalValue.Abs().StringFixed(int32(scale))
```

Comment:
```suggestion
		// Load the value into a fully padded (to precision and scale) string format, 
		// so that we can process the digit groups for the binary encoding.
		absStringVal := decimalValue.Abs().StringFixed(int32(scale))
```